### DISCUSSION TITLE
Mapping cables on the OpenStreetMap

### DISCUSSION THREAD

**Discussion by Vladyslavrz** [netbox-community/netbox#16380] [Ideas]
Add the ability to map cable lines on OpenStreetMap. Currently, we can add building coordinates. I suggest adding the ability to display buildings on the map, as well as the ability to draw lines (dotted, for example) between buildings, and also add junctions and various labels. This minimum set of features will greatly expand the capabilities of Netbox.

**Comment by pv2b** [comment]
There is a plugin for Netbox already that I think would do most of what you are requesting: https://github.com/drygdryg/netbox-plugin-device-map

The plugin uses OpenStreetMap to display devices and connections between those devices on a physical map just as you want. It does not directly support junctions and labels.

But you probably don't need special junction support, because you'll just end up placing at least one device in each junction, and the connection will be run between those devices.

It also does not support any way to do polylines between devices, everything's just represented as a straight line. Which is probably good enough for most reasonable uses of Netbox (it's not a surveying tool).

You may not need labels either if what the information needed can be inferred just from the device pins themselves.

I think the plugin might need a little love to support Netbox 4 in its current state (I haven't tried it), and if you want label support, that'll likely need to be added, but at least you wouldn't be starting from scratch.
